hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/ListKeyHandler.java

```diff
@@ -49,6 +49,15 @@ public class ListKeyHandler extends BucketSnapshotHandler {
   protected void execute(OzoneClient client, OzoneAddress address)
       throws IOException, OzoneClientException {
 
+    if (!Strings.isNullOrEmpty(address.getBucketName())) {
+      listKeysInsideBucket(client, address);
+      return;
+    }
+    listKeysInsideVolume(client, address);
+  }
+
+  private void listKeysInsideBucket(OzoneClient client, OzoneAddress address)
+      throws IOException {
     String volumeName = address.getVolumeName();
     String bucketName = address.getBucketName();
     String snapshotNameWithIndicator = address.getSnapshotNameWithIndicator();
```

Comment:
By default will show keys within snapshots as well or skip that in case of recursive commands?

Reply:
It will not list snapshots when listing at volume or bucket level.